ledger/ledger-configuration/src/test/suite/scala/com/daml/ledger/configuration/ConfigurationSpec.scala

```diff
@@ -0,0 +1,244 @@
+        val configurationBytes = protobuf.LedgerConfiguration
+          .of(
+            version = 1,
+            generation = 2,
+            timeModel = None,
+            maxDeduplicationTime = None,
+          )
+          .toByteArray
+
+        val configuration = Configuration.decode(configurationBytes)
+
+        configuration should be(
+          Left("Missing time model")
+        )
+      }
+    }
+
+    "decoding a v2 protobuf" should {
+      "decode a valid protobuf" in {
+        val configurationBytes = protobuf.LedgerConfiguration
+          .of(
+            version = 2,
+            generation = 3,
+            timeModel = Some(
+              protobuf.LedgerTimeModel.of(
+                avgTransactionLatency = Some(30.seconds.toProtobuf),
+                minSkew = Some(20.seconds.toProtobuf),
+                maxSkew = Some(5.minutes.toProtobuf),
+              )
+            ),
+            maxDeduplicationTime = Some(6.hours.toProtobuf),
+          )
+          .toByteArray
+
+        val configuration = Configuration.decode(configurationBytes)
+
+        configuration should be(
+          Right(
+            Configuration(
+              generation = 3,
+              timeModel = LedgerTimeModel(
+                avgTransactionLatency = 30.seconds.toJava,
+                minSkew = 20.seconds.toJava,
+                maxSkew = 5.minutes.toJava,
+              ).get,
+              maxDeduplicationTime = 6.hours.toJava,
+            )
+          )
+        )
+      }
+
+      "reject a missing time model" in {
+        val configurationBytes = protobuf.LedgerConfiguration
+          .of(
+            version = 2,
+            generation = 4,
+            timeModel = None,
+            maxDeduplicationTime = Some(1.day.toProtobuf),
+          )
+          .toByteArray
+
+        val configuration = Configuration.decode(configurationBytes)
+
+        configuration should be(
+          Left("Missing time model")
+        )
+      }
+
+      "reject a missing max deduplication time" in {
+        val configurationBytes = protobuf.LedgerConfiguration
+          .of(
+            version = 2,
+            generation = 1,
+            timeModel = Some(
+              protobuf.LedgerTimeModel.of(
+                avgTransactionLatency = Some(com.google.protobuf.duration.Duration.defaultInstance),
+                minSkew = Some(com.google.protobuf.duration.Duration.defaultInstance),
+                maxSkew = Some(com.google.protobuf.duration.Duration.defaultInstance),
+              )
+            ),
+            maxDeduplicationTime = None,
+          )
+          .toByteArray
+
+        val configuration = Configuration.decode(configurationBytes)
+
+        configuration should be(
+          Left("Missing maximum command time to live")
+        )
+      }
+
+      "rejects a negative transaction latency in the time model" in {
+        val configurationBytes = protobuf.LedgerConfiguration
```

Comment:
You may want to rewrite these test cases using table-driven property checks for easier readability (the test cases share most of the code).

Reply:
Good call. Done for the error cases.